internal/pkg/aws/ecs/ecs.go

```diff
@@ -168,6 +181,58 @@ func (e *ECS) ServiceTasks(clusterName, serviceName string) ([]*Task, error) {
 	return tasks, nil
 }
 
+// DefaultCluster returns the default clusters in the account and region.
+func (e *ECS) DefaultCluster() (string, error) {
+	resp, err := e.client.DescribeClusters(&ecs.DescribeClustersInput{})
+	if err != nil {
+		return "", fmt.Errorf("get default cluster: %w", err)
+	}
+
+	if len(resp.Clusters) == 0 {
+		return "", errors.New("no default cluster is found")
+	}
+
+	// NOTE: right now at most 1 default cluster is possible, so cluster[0] must be the default cluster
+	cluster := resp.Clusters[0]
+	return aws.StringValue(cluster.ClusterArn), nil
+}
+
+// RunTask runs a number of tasks with the task definition and network configurations in a cluster, and returns after
+// the task(s) is running or fails to run, along with task ARNs if possible.
+func (e *ECS) RunTask(input RunTaskInput) ([]string, error) {
+	resp, err := e.client.RunTask(&ecs.RunTaskInput{
+		Cluster:        aws.String(input.Cluster),
+		Count:          aws.Int64(int64(input.Count)),
+		LaunchType:     aws.String(ecs.LaunchTypeFargate),
+		StartedBy:      aws.String(input.StartedBy),
+		TaskDefinition: aws.String(input.TaskFamilyName),
+		NetworkConfiguration: &ecs.NetworkConfiguration{
+			AwsvpcConfiguration: &ecs.AwsVpcConfiguration{
+				AssignPublicIp: aws.String(ecs.AssignPublicIpEnabled),
+				Subnets:        aws.StringSlice(input.Subnets),
+				SecurityGroups: aws.StringSlice(input.SecurityGroups),
+			},
+		},
+	})
+	if err != nil {
+		return []string{}, fmt.Errorf("run task(s) %s: %w", input.TaskFamilyName, err)
+	}
+
+	taskArns := make([]string, len(resp.Tasks))
+	for idx, task := range resp.Tasks {
+		taskArns[idx] = aws.StringValue(task.TaskArn)
+	}
+
+	if err := e.client.WaitUntilTasksRunning(&ecs.DescribeTasksInput{
```

Comment:
Are there any specific errors that we should handle differently? I can imagine that if we fail to spin up a task maybe we can show a nicer error that our go errors.

"Failed to launch task with - essential container bla-bla-bla failed to start up" or something like that?

Reply:
From the doc I didn't find super helpful errors that are either common to all actions or are specific to this waiter https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_DescribeTasks.html... 

Do you think it might be helpful to direct the user to see the management console? In ECS > task > the specific task, it shows reasons why the task's been stopped under the Status field, which I think is not returned from the waiter API call.